proof/crefine/RISCV64/Ipc_C.thy

```diff
@@ -6082,6 +6082,18 @@ lemma schedContext_resume_ccorres:
                          to_bool_def)
   done
 
+lemma obj_at_cslift_ntfn:
+  fixes P :: "notification \<Rightarrow> bool"
+  shows "\<lbrakk>obj_at' P ntfn s; (s, s') \<in> rf_sr\<rbrakk> \<Longrightarrow>
+  \<exists>ko ko'. ko_at' ko ntfn s \<and> P ko \<and>
+        cslift s' (Ptr ntfn) = Some ko' \<and>
+        cnotification_relation (cslift s') ko ko'"
```

Comment:
Do you actually need the type annotation/`fixes`? I would have thought the `cnotification_relation` would constrain `ko`, which then should constrain `P`.

Reply:
I can check this. This lemma started off life as a copy paste of the analogous lemma for TCBs and was then modified. That’s where the fixes and type annotations would be from